### PR TITLE
Upgrade obeam to 0.1.1

### PR DESCRIPTION
**Pull request by amutake**
with some `failwith "not implemented"`

### CODE REVIEW

File: bin/main.ml [hunk 1]
```diff
@@ -38,7 +38,7 @@ let read_file beam_filename =
   try_with (fun () -> Bitstring.bitstring_of_file beam_filename) >>= fun beam ->
   beam_to_etf beam_filename beam >>= fun etf ->
   let sf = Simple_term_format.of_etf etf in
-  Ok (Abstract_format.of_sf sf)
+  Abstract_format.of_sf sf |> map_error ~f:(fun e -> Failure (Abstract_format.sexp_of_err_t e |> Sexp.to_string))
```
Comment (amutake) [inline bin/main.ml]: because `Abstract_format.err_t` is not an instance of `exn`

File: bin/main.ml [hunk 2]
```diff
@@ -5,23 +5,23 @@ open Fialyzer
 
 let extract_debug_info_buf filename layout =
   let {
-    Chunk.cl_abst = opt_abst;
-    Chunk.cl_dbgi = opt_dbgi;
+    Beam.cl_abst = opt_abst;
```
Comment (amutake) [inline bin/main.ml]: module renamed: `Chunk` to `Beam`

File: lib/erl_type.ml [hunk 3]
```diff
@@ -31,7 +31,7 @@ type t_map_mandatoriness = Mandatory | Optional
 let mandatoriness_of_etf = function
   | Etf.Atom "mandatory" -> Ok Mandatory
   | Atom "optional" -> Ok Optional
-  | other -> Error (Failure (!%"mandatoriness_of_etf error: %s" (Etf.show other)))
+  | other -> Error (Failure (!%"mandatoriness_of_etf error: %s" (E.show_etf other)))
```
Comment (amutake) [inline lib/erl_type.ml]: From `0.1.0`, `Etf.t` is not an instance of `show`.

File: lib/from_erlang.ml [hunk 4]
```diff
@@ -31,13 +39,26 @@ let rec expr_of_erlang_expr = function
...
+     (* TODO: fun _(...) -> ... end *)
+     failwith "not implemented"
   | ExprLocalCall (_line_t, f, args) ->
      App (expr_of_erlang_expr f, List.map ~f:expr_of_erlang_expr args)
   | ExprRemoteCall (_line_t, _line_m, m, f, args) ->
      let mfa = MFA (expr_of_erlang_expr m, expr_of_erlang_expr f, Constant (Number (List.length args))) in
      App (mfa, List.map ~f:expr_of_erlang_expr args)
+  | ExprMatch (line_t, pat, e) ->
+     (* HACK: `A = B` convert to case expr `case B of A -> B end` *)
+     expr_of_erlang_expr (F.ExprCase (line_t, e, [F.ClsCase (line_t, pat, None, e)]))
```
Comment (yoshihiro503) [inline lib/from_erlang.ml]: ## It's Ninja Pattern Matching. 

Can the following expressions see the bound variables in pattern `A` ?

For example,


```erlang
{Name, Age} = foo(),
<expr>
```

is equals to:

```erlang
case foo() of
  {Name, Age} ->
    <expr>
end
```

but not equals to:
```erlang
case foo() of
  {Name, Age} ->
    foo()
end,
<expr>
```

File: lib/from_erlang.ml [hunk 5]
```diff
@@ -31,13 +39,26 @@ let rec expr_of_erlang_expr = function
     | F.ClsCase (_, _, _, _) | F.ClsFun (_, _, _, _) -> failwith "not implemented"
     ) in
     Case (expr_of_erlang_expr e, cs)
+  | ExprLocalFunRef (_line_t, name, arity) ->
+     (* TODO: support local `fun F/A` *)
+     failwith "not implemented"
```
Comment (yoshihiro503) [inline lib/from_erlang.ml]: Please make a issue.

Comment (yoshihiro503) [reply lib/from_erlang.ml]: and
```ocaml
raise Known_error.(FialyzerError (NotImplemented issue))
```

File: lib/from_erlang.ml (hunk 4 above)
Comment (amutake) [reply lib/from_erlang.ml]: You're right!
Then I think it is needed to add some extensions into `Ast_intf.expr`.
I'll create issue and replace the code with `NotImplemented` error.

File: lib/from_erlang.ml (hunk 5 above)
Comment (amutake) [reply lib/from_erlang.ml]: OK. I'll do that.

File: lib/from_erlang.ml (hunk 4 above)
Comment (amutake) [reply lib/from_erlang.ml]: https://github.com/dwango/fialyzer/pull/78/commits/0e84ef82149c15a23a3da4203d5092029db76693

File: lib/from_erlang.ml (hunk 5 above)
Comment (amutake) [reply lib/from_erlang.ml]: https://github.com/dwango/fialyzer/pull/78/commits/0e84ef82149c15a23a3da4203d5092029db76693

**Comment by amutake** [comment]
Thank you 🙏

**Pull request merged**